### PR TITLE
Wrap IsPodEnrolledInDp for iptables and ebpf mode

### PR DESCRIPTION
**Pull request by PlatformLC**
**Please provide a description of this PR:**

massive unexpected error log in istio-cni while using eBPF redirect mode
```
│ 2023-11-17T06:54:58.937501Z    error    ambient    Failed to list ipset entries: failed to list ipset ztunnel-pods-ips: no such file or directory                                                                                                                                 │
│ 2023-11-17T06:55:00.951899Z    error    ambient    Failed to list ipset entries: failed to list ipset ztunnel-pods-ips: no such file or directory
```
In eBPF mode, ipset is not involved, thus wraps to `IsPodEnrolledInDp` for both mode instead of `IsPodInIpset`

### CODE REVIEW

File: cni/pkg/ambient/net_linux.go [hunk 1]
```diff
@@ -1326,6 +1326,16 @@ func (s *Server) getEnrolledIPSets() sets.String {
 	return pods
 }
 
+func (s *Server) IsPodEnrolledInDp(pod *corev1.Pod) bool {
```
Comment (hzxuzhonghu) [inline cni/pkg/ambient/net_linux.go:1329]: What does DP mean? DataPlane?

Comment (PlatformLC) [reply cni/pkg/ambient/net_linux.go:1329]: Yes. Dp represents Dataplane.

File: cni/pkg/ebpf/server/redirectionServer_linux.go [hunk 2]
```diff
@@ -686,6 +686,17 @@ func (r *RedirectServer) DumpAppIPs() sets.String {
 	return m
 }
 
+func (r *RedirectServer) IsPodIPEnrolled(ip string) bool {
+	var valueOut mapInfo
+	ipAddr, err := netip.ParseAddr(ip)
+	if err != nil {
+		return false
+	}
+	err = r.obj.AppInfo.Lookup(ipAddr.As4(), &valueOut)
```
Comment (hzxuzhonghu) [inline cni/pkg/ebpf/server/redirectionServer_linux.go:695]: nit: return r.obj.AppInfo.Lookup == nil

Does it only return err when the IP not found?

File: cni/pkg/ambient/net_linux.go (hunk 1 above)
Comment (hzxuzhonghu) [reply cni/pkg/ambient/net_linux.go:1329]: so rename to `IsPodEnrolledInMesh` or specificlly `IsPodEnrolledInAmbient`

File: cni/pkg/ebpf/server/redirectionServer_linux.go (hunk 2 above)
Comment (PlatformLC) [reply cni/pkg/ebpf/server/redirectionServer_linux.go]: Updated

**Review by bleggett — APPROVED** [review]
LGTM - combining them into one call makes sense, they're both fundamentally IP maps.

File: cni/pkg/ebpf/server/redirectionServer_linux.go [hunk 3]
```diff
@@ -131,7 +131,7 @@ func EBPFTProxySupport() bool {
 		return true
 	}
 	if errors.Is(err, ebpf.ErrNotSupported) {
-		log.Infof("FnSkAssign (Linux 5.7 or later) is not supported in current kernel")
+		log.Debugf("FnSkAssign (Linux 5.7 or later) is not supported in current kernel")
```
Comment (bleggett) [inline cni/pkg/ebpf/server/redirectionServer_linux.go:134]:
```suggestion
		log.Debugf("FnSkAssign (Linux 5.7 or later) is not supported in current kernel")
```

IMO this should be WARN - it's something you would want to know even if you hadn't explicitly enabled debug logging.

**Review by bleggett — APPROVED** [review]

**Review by hzxuzhonghu — APPROVED** [review]

**Pull request merged**